src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1468,7 +1449,11 @@ def forward(
         if output_hidden_states:
             all_hidden_states += (hidden_states,)
 
-        next_cache = next_decoder_cache if use_cache else None
+        next_cache = past_key_values if use_cache else None
+        if return_self_attention_cache:
+            next_cache = past_key_values.self_attention_cache
+        if return_legacy_cache:
+            next_cache = past_key_values.to_legacy_cache()
```

Comment:
let's deprecate here as welll!

Reply:
Added a deprecation warning for v4.43 https://github.com/huggingface/transformers/pull/31166/commits/1860c315beb565f5e3b7cee9ab5f012dafd17ac9